addons/account/models/chart_template.py

```diff
@@ -295,9 +296,10 @@ def tax_template_changed(tax, template):
                 elif model_name == 'account.account':
                     # Point or create xmlid to existing record to avoid duplicate code
                     account = self.ref(xmlid, raise_if_not_found=False)
-                    if not account or (account and account.code != values['code']):
+                    normalized_code = f'{values["code"]:<0{code_digits}}'
+                    if not account or (account and account.code != normalized_code):
                         existing_account = self.env['account.account'].search([
-                            ('code', '=', values['code']),
+                            ('code', '=', normalized_code),
```

Comment:
Maybe something like this?
```suggestion
                            ('code', '=like', values['code'] + "%"),
```

The best would be with a regex...
The easiest way (but not super elegant) would be to use `_search` with `add_where`.

```python
query = self.env['account.account']._search([('company_id', '=', company.id)])
query.add_where("account_account.code SIMILAR TO '%s0*'", [values['code']])
existing_account = self.env['account.account'].browse(query)
```

Reply:
I've made the changes. It's not very elegant... but it seems to get the job done.